### PR TITLE
IGNITE-21558 Sql. Remove ExecutionContext dependency from ExpressionFactory

### PR DESCRIPTION
**Pull request by korlov42**
https://issues.apache.org/jira/browse/IGNITE-21558

To be sure this patch doesn't affect performance, I've run a few benchmarks. Results as follow:

```
Before
Benchmark                          (clusterSize)  (fsync)                                    Mode  Cnt    Score   Error  Units
SelectBenchmark.sqlGet                         1    false                                    avgt   20    5.604 ± 0.098  us/op

Benchmark                          (clusterSize)  (fsync)  (partitionCount)  (replicaCount)  Mode  Cnt    Score   Error  Units
InsertBenchmark.sqlInlinedInsert               1    false                24               1  avgt   20  134.026 ± 3.023  us/op
InsertBenchmark.sqlPreparedInsert              1    false                24               1  avgt   20   44.318 ± 8.932  us/op


After 
Benchmark                          (clusterSize)  (fsync)                                    Mode  Cnt    Score   Error  Units
SelectBenchmark.sqlGet                         1    false                                    avgt   20    5.639 ± 0.087  us/op

Benchmark                          (clusterSize)  (fsync)  (partitionCount)  (replicaCount)  Mode  Cnt    Score   Error  Units
InsertBenchmark.sqlInlinedInsert               1    false                24               1  avgt   20  130.579 ± 1.622  us/op
InsertBenchmark.sqlPreparedInsert              1    false                24               1  avgt   20   43.550 ± 6.111  us/op
```
----------
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/AccumulatorsFactory.java [hunk 1]
```diff
@@ -267,11 +265,13 @@ private final class AccumulatorWrapperImpl implements AccumulatorWrapper<RowT> {
         private final boolean distinct;
 
         AccumulatorWrapperImpl(
+                RowHandler<RowT> handler,
```
Comment (AMashenkov) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/AccumulatorsFactory.java:268]: If captured context is no longer used, can this class be static?

**Review by AMashenkov — APPROVED** [review]

**Review by xtern — APPROVED** [review]

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/AccumulatorsFactory.java (hunk 1 above)
Comment (korlov42) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/AccumulatorsFactory.java:276]: good point, fixed

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/AccumulatorsFactory.java [hunk 2]
```diff
@@ -57,7 +57,7 @@
  * AccumulatorsFactory.
  * TODO Documentation https://issues.apache.org/jira/browse/IGNITE-15859
  */
-public class AccumulatorsFactory<RowT> implements Supplier<List<AccumulatorWrapper<RowT>>> {
+public class AccumulatorsFactory<RowT> {
```
[the remark this answers is not in the document]
Comment (korlov42) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/AccumulatorsFactory.java:66]: javadoc updated

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/AccumulatorsFactory.java [hunk 3]
```diff
@@ -142,26 +142,24 @@ private static Function<Object, Object> compileCast(IgniteTypeFactory typeFactor
      * TODO Documentation https://issues.apache.org/jira/browse/IGNITE-15859
      */
     public AccumulatorsFactory(
-            ExecutionContext<RowT> ctx,
```
[the remark this answers is not in the document]
Comment (korlov42) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/AccumulatorsFactory.java:145]: done

**Pull request merged**